velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -370,6 +378,30 @@ TEST_F(StringTest, endsWith) {
   EXPECT_EQ(endsWith(std::nullopt, "abc"), std::nullopt);
 }
 
+TEST_F(StringTest, substring_index) {
+  EXPECT_EQ(substring_index("www.apache.org", ".", 3), "www.apache.org");
+  EXPECT_EQ(substring_index("www.apache.org", ".", 2), "www.apache");
+  EXPECT_EQ(substring_index("www.apache.org", ".", 1), "www");
+  EXPECT_EQ(substring_index("www.apache.org", ".", 0), "");
+  EXPECT_EQ(substring_index("www.apache.org", ".", -1), "org");
+  EXPECT_EQ(substring_index("www.apache.org", ".", -2), "apache.org");
+  EXPECT_EQ(substring_index("www.apache.org", ".", -3), "www.apache.org");
+  // str is empty string
```

Comment:
Let's capitalize the first letter and add `.` at the end for comments.

Reply:
Done.